### PR TITLE
fix case-insensitive keyword query and consistent state for alerting

### PR DESCRIPTION
**Pull request by junotx**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

let keyword query case-insensitive for alerting 
let consistent for listing and geting rules

**Which issue(s) this PR fixes**:

commit https://github.com/kubesphere/kubesphere/pull/3674/commits/68b0b8f4ba3614623d27f0f83881727c88fdf279 fixes https://github.com/kubesphere/kubesphere/issues/3648
commit https://github.com/kubesphere/kubesphere/pull/3674/commits/66c0e0dad1475d1cc52a34af124a681f7d880ce5 fixes consistent state for listing and geting rules

/cc @benjaminhuo

### CODE REVIEW

File: pkg/api/alerting/v2alpha1/types.go [hunk 1]
```diff
@@ -526,3 +526,10 @@ func NewBulkItemErrorServerResponse(ruleName string, err error) *BulkItemRespons
 		Error:     err,
 	}
 }
+
+// containsI reports whether substr is case-insensitive within s.
+func containsI(s, substr string) bool {
```
Comment (zryfish) [inline pkg/api/alerting/v2alpha1/types.go:531]: `containsI` is not a readable function name, how about `containsCaseInsensitive`

**Comment by benjaminhuo** [comment]
Pls change the Github email address to the yunify one from now on

File: pkg/api/alerting/v2alpha1/types.go (hunk 1 above)
Comment (benjaminhuo) [reply pkg/api/alerting/v2alpha1/types.go:531]: Do we really need a new function here?

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**